scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -1302,6 +1302,28 @@ def test_mean(xp, ndim: int):
         xp_assert_close(r.mean().magnitude(), desired, atol=atol)
 
 
+@make_xp_test_case(Rotation.from_rotvec, Rotation.mean, Rotation.magnitude)
+@pytest.mark.parametrize("ndim", range(1, 5))
+def test_mean_axis(xp, ndim: int):
```

Comment:
The expected mean being zero in all these cases is a useful check, but doesn't fully exercise things IMO. Could you add a test with random rotations, that compares a reduction along several axes to a list of axis=None mean of rotations formed from those slices? That will prove out the equivalence to the existing algorithm.

Reply:
Added a dedicated test for it. Let me know if you had something else in mind!